Update Box Connector with technical changes (#338)
mulesoft/docs-connectors: modules/ROOT/pages/box/box-connector-config-topics.adoc

endif::[]
:imagesdir: ../../assets/images/

> To use the Box connector in a Mule application, configure a global Box configuration element that can be used by as many Box connectors as needed for your application.

Reviewer: ```suggestion
To use the Box connector in a Mule app, configure a global Box configuration element to use by as many Box connectors as needed for your application.
```

== Configure the Box Connector Global Element

. Click the **Global Elements** tab at the base of the canvas.
. In the *Global Configuration Elements* screen, click *Create*.
. In the *Choose Global Type* wizard, expand Connector Configuration and select *Mule-box-connector Config* and click *OK*.

> . In the configuration screen for `OAUTH 2.0` with Authorization Code, grant type displays. This is the only configuration that Box connector supports.

Reviewer: This doesn't seem like a step. It seems like a step result.  I don't see the user taking any action here.  ?

. Configure the properties as described here.
+
[%header,cols="30a,70a"]
|===
|Parameter|Description
|`clientId`|Specifies the client identifier as assigned by the authorization server when the client application was registered.
|`clientSecret`|Specifies the client application's client secret.
|`host`|The host where Box exposes its API.
|`port`|The port where Box services listen for incoming requests. Default: 443
|`basePath`|The base path for the Box services' base URI.
|`protocol`|The protocol which Box services follow. Default: HTTPS
|`localCallbackPath`|The local callback path. Default: /callback
|`localCallbackConfig`|The HTTP Listener configuration that is used to configure the local authorization URL.
|`externalCallbackUrl`|The redirect URL that is configured in the client application.
|`localAuthorizationUrl`|Local authorization URL that is used to initiate an `OAuth2` dance.
|`authorizationUrl`|The authorization URL to request for an authorization code.

> |`accessTokenUrl`|Specifies the URL for requesting for an access token.

Reviewer: ```suggestion
|`accessTokenUrl`|Specifies the URL for requesting an access token.
```

|===
+
In the `OAUTH 2.0` configuration screen, the placeholder values refer to a configuration file `mule-artifact.properties` placed in the
`src/main/resources` folder of your project.
+
You can either enter your credentials into the global configuration properties, or reference a configuration file that contains these values.
+

> NOTE: For simpler maintenance and better re-usability of your project, Mule recommends that you use a configuration file.

Reviewer: ```suggestion
NOTE: Use a configuration file for simpler maintenance and better reusability of your project.
```

> Keeping these values in a separate file is useful if you need to deploy to different environments such as: production, development, and QA (where your access credentials are different).

Reviewer: ```suggestion
Keeping these values in a separate file is useful if you need to deploy to different environments where your access credentials are different, such as production, development, and QA.
```
